flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adapter/DefaultExecutionVertexTest.java

```diff
@@ -18,131 +18,112 @@
 
 	private List<SchedulingExecutionVertex> schedulingExecutionVertices;
 
-	private List<ExecutionVertex> executionVertices;
+	private IntermediateResultPartitionID intermediateResultPartitionId;
 
 	@Before
 	public void setUp() throws Exception {
-		JobVertex[] jobVertices = new JobVertex[2];
-		jobVertices[0] = createNoOpVertex(parallelism);
-		jobVertices[1] = createNoOpVertex(parallelism);
-		jobVertices[1].connectNewDataSetAsInput(jobVertices[0], ALL_TO_ALL, BLOCKING);
-		jobVertices[0].setInputDependencyConstraint(ALL);
-		jobVertices[1].setInputDependencyConstraint(ANY);
-		ExecutionGraph executionGraph = createSimpleTestGraph(
-			new JobID(),
-			taskManagerGateway,
-			triggeredRestartStrategy,
-			jobVertices);
-		ExecutionGraphToSchedulingTopologyAdapter adapter = new ExecutionGraphToSchedulingTopologyAdapter(executionGraph);
-
-		schedulingExecutionVertices = new ArrayList<>();
-		adapter.getVertices().forEach(vertex -> schedulingExecutionVertices.add(vertex));
-		executionVertices = new ArrayList<>();
-		executionGraph.getAllExecutionVertices().forEach(vertex -> executionVertices.add(vertex));
-	}
 
-	@Test
-	public void testGetId() {
-		for (int idx = 0; idx < schedulingExecutionVertices.size(); idx++){
-			assertEquals(schedulingExecutionVertices.get(idx).getId().getJobVertexId(),
-				executionVertices.get(idx).getJobvertexId());
-			assertEquals(schedulingExecutionVertices.get(idx).getId().getSubtaskIndex(),
-				executionVertices.get(idx).getParallelSubtaskIndex());
-		}
+		schedulingExecutionVertices = new ArrayList<>(2);
+		intermediateResultPartitionId = new IntermediateResultPartitionID();
+
+		DefaultResultPartition schedulingResultPartition = new DefaultResultPartition(
+			intermediateResultPartitionId,
+			new IntermediateDataSetID(),
+			BLOCKING);
+		DefaultExecutionVertex schedulingVertex1 = new DefaultExecutionVertex(
+			new ExecutionVertexID(new JobVertexID(), 0),
+			Collections.singletonList(schedulingResultPartition),
+			ALL,
+			stateProvider);
+		schedulingResultPartition.setProducer(schedulingVertex1);
+		DefaultExecutionVertex schedulingVertex2 = new DefaultExecutionVertex(
+			new ExecutionVertexID(new JobVertexID(), 0),
+			Collections.emptyList(),
+			ALL,
+			stateProvider);
+		schedulingVertex2.addConsumedPartition(schedulingResultPartition);
+		schedulingExecutionVertices.add(schedulingVertex1);
+		schedulingExecutionVertices.add(schedulingVertex2);
 	}
 
 	@Test
 	public void testGetExecutionState() {
 		final ExecutionState[] states = ExecutionState.values();
-		Random random = new Random();
-		for (ExecutionVertex executionVertex: executionVertices) {
-			setVertexState(executionVertex, states[Math.abs(random.nextInt()) % states.length]);
-		}
-
-		for (int idx = 0; idx < schedulingExecutionVertices.size(); idx++) {
-			assertEquals(schedulingExecutionVertices.get(idx).getState(), executionVertices.get(idx).getExecutionState());
+		for (ExecutionState state : states) {
```

Comment:
could inline `ExecutionState.values()` to `for (ExecutionState state : ExecutionState.values()) {`

Reply:
OK